### PR TITLE
[onert] Add BulkLayer on TRIX backend

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit adds BulkLayer on TRIX backend.
It uses trix-engine library to handle TRIX device.

ONE-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

draft pr: #8805

### CODE REVIEW

File: runtime/onert/backend/trix/DevContext.h [hunk 1]
```diff
@@ -55,6 +55,64 @@ class DevContext
...
+        return DATA_LAYOUT_NHWC;
+      default:
+        throw std::runtime_error("Unknown Layout");
+    }
+  }
+
+  data_type convertDataType(const ir::DataType type)
+  {
+    switch (type)
+    {
```
Comment (jyoungyun) [inline runtime/onert/backend/trix/DevContext.h:98]: I need to consider another case more later(?).

Comment (jyoungyun) [reply runtime/onert/backend/trix/DevContext.h:98]: Is it okay? :) /cc @hseok-oh

Comment (hseok-oh) [reply runtime/onert/backend/trix/DevContext.h:98]: I prefer to support type conversion which can test, and remain NYI others.

File: runtime/onert/backend/trix/DevContext.h [hunk 2]
```diff
@@ -55,6 +55,64 @@ class DevContext
 
   npudev_h getDev() { return _dev_handle; }
 
+  template <typename T> void setDataInfo(tensors_data_info *info, std::vector<T *> &tensors)
```
Comment (ragmani) [inline runtime/onert/backend/trix/DevContext.h:58]: I think there is no reason to use `function template`

File: runtime/onert/backend/trix/DevContext.h [hunk 3]
```diff
@@ -55,6 +55,64 @@ class DevContext
...
+    info->num_info = static_cast<uint32_t>(tensors.size());
+
+    for (uint32_t idx = 0; idx < info->num_info; ++idx)
+    {
+      info->info[idx].layout = convertDataLayout(tensors[idx]->layout());
+      info->info[idx].type = convertDataType(tensors[idx]->data_type());
+    }
+  }
+
+  template <typename T> void setBuffer(generic_buffers *buf, std::vector<T *> &tensors)
```
Comment (ragmani) [inline runtime/onert/backend/trix/DevContext.h:69]: Here too

File: runtime/onert/backend/trix/DevContext.h (hunk 2 above)
Comment (jyoungyun) [reply runtime/onert/backend/trix/DevContext.h:58]: This function supports both case of `const IPortableTensor` as input and `IPortableTensor` as output. So I use template function.

File: runtime/onert/backend/trix/DevContext.h (hunk 3 above)
Comment (jyoungyun) [reply runtime/onert/backend/trix/DevContext.h:69]: ditto. :)

File: runtime/onert/backend/trix/DevContext.h (hunk 2 above)
Comment (ragmani) [reply runtime/onert/backend/trix/DevContext.h:58]: I think that a function `void setDataInfo(tensors_data_info *info, const std::vector<const IPortableTensor *> &tensors)` supports both cases. Is there a problem?

Comment (jyoungyun) [reply runtime/onert/backend/trix/DevContext.h:58]: I don't think that const Variable will be compatible with Variable........ Am I wrong..?

```
/opt/test/runtime/onert/backend/trix/ops/BulkLayer.cc:85:40: error: cannot convert 'std::vector<onert::backend::IPortableTensor*>' to 'const std::vector<const onert::backend::IPortableTensor*>&'
   85 |   _dev_context->setDataInfo(&out_info, _outputs);
      |                                        ^~~~~~~~
      |                                        |
      |                                        std::vector<onert::backend::IPortableTensor*>
In file included from /opt/test/runtime/onert/backend/trix/ops/BulkLayer.h:21,
                 from /opt/test/runtime/onert/backend/trix/ops/BulkLayer.cc:17:
/opt/test/runtime/onert/backend/trix/ops/../DevContext.h:58:89: note:   initializing argument 2 of 'void onert::backend::trix::DevContext::setDataInfo(tensors_data_info*, const std::vector<const onert::backend::IPortableTensor*>&)'
   58 |   void setDataInfo(tensors_data_info *info, const std::vector<const IPortableTensor *> &tensors)
      |                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
make[3]: *** [runtime/onert/backend/trix/CMakeFiles/onert_backend_trix.dir/build.make:102: runtime/onert/backend/trix/CMakeFiles/onert_backend_trix.dir/ops/BulkLayer.cc.o] Error 1
```

Comment (ragmani) [reply runtime/onert/backend/trix/DevContext.h:58]: I did't know they are different types. How about using a way (template <class const_iter>
void f(const_iter b, const_iter e) {} ) in [link](https://code-examples.net/en/q/2013e4). It's not madatory, so you don't have to.

Comment (jyoungyun) [reply runtime/onert/backend/trix/DevContext.h:58]: I've viewed the link, but the `setDataInfo` function requires a total size of the tensor vector. If I pass the const_iterator of the vector, I have to assign the total tensor size outside the `setDataInfo` function. It is not a pretty code I think. Why don't we just keep it?

**Comment by jyoungyun** [comment]
PTAL

File: runtime/onert/backend/trix/DevContext.h (hunk 2 above)
Comment (ragmani) [reply runtime/onert/backend/trix/DevContext.h:58]: OK. The suggested way doesn't look good either.

**Review by ragmani — APPROVED** [review]
LGTM

**Review by hseok-oh — APPROVED** [review]

**Pull request merged**